src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -43,308 +25,69 @@ partial class BorderLayerRenderer
 		}
-		);
 
-		compositor.InvalidateRender(visual);
-
-		return disposables;
-	}
-
-	private static void CreateImageLayer(Compositor compositor, CompositeDisposable disposables, Thickness borderThickness, Rect adjustedArea, CompositionSpriteShape backgroundShape, Rect backgroundArea, ImageBrush imgBackground)
-	{
-		Action onInvalidateRender = () =>
+		// Border shape (if any)
+		_borderShapeBrushDisposable.Disposable = null;
+		if (borderThickness != Thickness.Empty)
 		{
-			if (imgBackground.ImageDataCache is not { } imageData)
-			{
-				return;
-			}
-
-			if (imageData.Error is null)
-			{
-				var surfaceBrush = compositor.CreateSurfaceBrush(imageData.CompositionSurface);
-
-				var sourceImageSize = new Size(imageData.CompositionSurface.Image.Width, imageData.CompositionSurface.Image.Height);
-
-				// We reduce the adjustedArea again so that the image is inside the border (like in Windows)
-				var imageArea = adjustedArea.DeflateBy(borderThickness);
-
-				backgroundArea = imgBackground.GetArrangedImageRect(sourceImageSize, imageArea);
-
-				// surfaceBrush.Offset = new Vector2((float)imageFrame.Left, (float)imageFrame.Top);
-				var matrix = Matrix3x2.CreateScale((float)(backgroundArea.Width / sourceImageSize.Width), (float)(backgroundArea.Height / sourceImageSize.Height));
-				matrix *= Matrix3x2.CreateTranslation((float)backgroundArea.Left, (float)backgroundArea.Top);
-
-				if (imgBackground.Transform != null)
-				{
-					matrix *= imgBackground.Transform.MatrixCore;
-				}
-
-				surfaceBrush.TransformMatrix = matrix;
-
-				backgroundShape.FillBrush = surfaceBrush;
-			}
-			else
-			{
-				backgroundShape.FillBrush = null;
-			}
-		};
-
-		onInvalidateRender();
-		imgBackground.InvalidateRender += onInvalidateRender;
-		new DisposableAction(() => imgBackground.InvalidateRender -= onInvalidateRender).DisposeWith(disposables);
-	}
-
-	private static CompositionPath GetBackgroundPath(SKPoint[] radii, Rect area)
-	{
-		var geometrySource = new SkiaGeometrySource2D();
-		SetRoundedPath(geometrySource.Geometry, area.ToSKRect(), radii);
-
-		return new CompositionPath(geometrySource);
-	}
-
-	private static CompositionPath GetBorderPath(SKPoint[] outerRadii, SKPoint[] innerRadii, Rect outerArea, Rect innerArea)
-	{
-		var geometrySource = new SkiaGeometrySource2D();
-		SetRoundedPath(geometrySource.Geometry, outerArea.ToSKRect(), outerRadii);
-		SetRoundedPath(geometrySource.Geometry, innerArea.ToSKRect(), innerRadii);
-		geometrySource.Geometry.FillType = SKPathFillType.EvenOdd;
-
-		return new CompositionPath(geometrySource);
-	}
-
-	private static void SetRoundedPath(in SKPath geometry, SKRect area, SKPoint[] radii)
-	{
-		var roundRect = new SKRoundRect();
-		roundRect.SetRectRadii(area, radii);
-		geometry.AddRoundRect(roundRect);
-		geometry.Close();
+			var borderShape = visual.BorderShape;
+			var borderBrush = state.BorderBrush;
```

Comment:
I would assume `borderBrush` can be `null` and include it with the `borderThickness != Thickness.Empty` check.